ndarray-rand/src/lib.rs

```diff
@@ -141,6 +246,39 @@ where
     {
         Self::from_shape_fn(shape, |_| dist.sample(rng))
     }
+
+    fn sample_axis(&self, axis: Axis, n_samples: usize, with_replacement: bool) -> Array<A, D>
+    where
+        A: Copy,
+        D: RemoveAxis,
+    {
+        self.sample_axis_using(axis, n_samples, with_replacement, &mut get_rng())
+    }
+
+    fn sample_axis_using<R>(
+        &self,
+        axis: Axis,
+        n_samples: usize,
+        with_replacement: bool,
+        rng: &mut R,
+    ) -> Array<A, D>
+    where
+        R: Rng + ?Sized,
+        A: Copy,
+        D: RemoveAxis,
+    {
+        let indices: Vec<_> = if with_replacement {
+            let distribution = Uniform::from(0..self.len_of(axis));
+            (0..n_samples).map(|_| distribution.sample(rng)).collect()
+        } else {
+            sample(rng, self.len_of(axis), n_samples).into_vec()
+        };
+        self.select(Axis(0), &indices)
```

Comment:
should not be Axis(0)

Reply:
Now, maybe axis should be a parameter to the quickcheck tests

Reply:
Rookie mistake, and having it as a test parameter would have indeed caught it.